std_execution.bs

```diff
@@ -4555,6 +4567,71 @@ from the operation before the operation completes.
 
         - Otherwise, <code>execution::set_done(<i>GET-BASE</i>(static_cast&lt;Derived&amp;&amp;>(self)))</code>
 
+### `execution::receiver_signatures` <b>[execution.snd_rec_utils.rcvr_sigs]</b> ### {#spec-execution.snd_rec_utils.rcvr_sigs}
+
+1. `receiver_signatures` is used to define a type that implements the nested `value_types`, `error_types`, and `sends_done` members that describe a typed sender. Its arguments are a flat list of function types that describe the signatures of the receiver's completion-signal operations that the sender invokes.
+
+2. [<i>Example:</i>
+     <pre highlight="c++">
+     using my_traits =
+       execution::receiver_signatures&lt;
+         execution::set_value_t(),
+         execution::set_value_t(int, float),
+         execution::set_error_t(exception_ptr),
+         execution::set_error_t(error_code),
+         execution::set_done_t()
+       >;
+
+      // my_traits::value_types&lt;tuple, variant> names the type:
+      //   variant&lt;tuple&lt;>, tuple&lt;int, float>>
+      //
+      // my_traits::error_types&lt;variant> names the type:
+      //   variant&lt;exception_ptr, error_code>
+      //
+      // my_traits::sends_done is true
+     </pre>
+     -- <i>end example</i>]
+
+3. This section makes use of the following exposition-only concept:
+
+    <pre highlight="c++">
+    template &lt;class Fn>
+      concept <i>receiver-signature</i> = <i>see below</i>;
+    </pre>
+
+    1. A type `Fn` satisfies <code><i>receiver-signature</i></code> if it is a function type with one of the following forms:
+
+        * <code>set_value_t(<i>Vs</i>...)</code>, where <code><i>Vs</i></code> is an arbitrary parameter pack.
+        * <code>set_error_t(<i>E</i>)</code>, where <code><i>E</i></code> is an arbitrary type.
+        * `set_done_t()`
+
+    2. Otherwise, `Fn` does not satisfy <code><i>receiver-signature</i></code>.
+
+4.  <pre highlight="c++">
+    template &lt;<i>receiver-signature</i>... Fns>
```

Comment:
Should this be later extended to support listing other queries on the receiver?
e.g. that it needs to implement `get_scheduler()`?

Reply:
Well, it's not really necessary. This utility is for providing properties of the _sender_ that are not otherwise inspectable. The only time you would need information about the receiver queries is when type-erasing the sender, in which case we could have a sister utility: `receiver_queries`.

Reply:
The other reason I would like to avoid putting receiver queries in here is that the syntax for the completion signals would have to change to something like this:

```c++
using _traits = completion_signatures<
  void(set_value_t, Vals0...),
  void(set_value_t, Vals1...),
  void(set_value_t, Vals2...),
  ...
  void(set_error_t, Err0) noexcept,
  void(set_error_t, Err1) noexcept,
  ...
  void(set_done_t) noexcpet
>;
```

... which I don't consider to be an improvement.

Also, then the name `completion_signatures` no longer makes sense. It would have to be something like `receiver_requirements` or some such, and even that would stop making sense if we go the direction of #320 and move the receiver queries into a separate context object.